beacon-chain/p2p/peers/peerdata/store.go

```diff
@@ -0,0 +1,100 @@
+package peerdata
+
+import (
+	"context"
+	"errors"
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum/p2p/enr"
+	"github.com/libp2p/go-libp2p-core/network"
+	"github.com/libp2p/go-libp2p-core/peer"
+	ma "github.com/multiformats/go-multiaddr"
+	ethpb "github.com/prysmaticlabs/ethereumapis/eth/v1"
+	pb "github.com/prysmaticlabs/prysm/proto/beacon/p2p/v1"
+)
+
+var (
+	// ErrPeerUnknown is returned when there is an attempt to obtain data from a peer that is not known.
+	ErrPeerUnknown = errors.New("peer unknown")
+)
+
+// PeerConnectionState is the state of the connection.
+type PeerConnectionState ethpb.ConnectionState
+
+// StoreConfig holds peer store parameters.
+type StoreConfig struct {
+	MaxPeers int
+}
+
+// Store is a container for various peer related data (both protocol and app level).
+// Container implements RWMutex, so data access can be restricted on the container level. This allows
+// different components rely on the very same peer map container.
+type Store struct {
+	sync.RWMutex
+	ctx    context.Context
+	config *StoreConfig
+	peers  map[peer.ID]*PeerData
+}
+
+// PeerData aggregates protocol and application level info about a single peer.
+type PeerData struct {
+	// Network related data.
+	Address   ma.Multiaddr
+	Direction network.Direction
+	ConnState PeerConnectionState
+	Enr       *enr.Record
+	// Chain related data.
+	ChainState            *pb.Status
+	MetaData              *pb.MetaData
+	ChainStateLastUpdated time.Time
+	// Scorers related data.
+	BadResponses         int
+	ProcessedBlocks      uint64
+	BlockProviderUpdated time.Time
+}
+
+// NewStore creates new peer data store.
+func NewStore(ctx context.Context, config *StoreConfig) *Store {
```

Comment:
Appropriate locks need to be applied to all the receiver methods

Reply:
Responded in discord:
> the store is used by different packages, and they control access to its contents (the sole purpose of store is to share access to mutex - so that different packages can lock, while store package itself doesn't lock - to avoid double lock). If you look at all the methods that are using store, they start with `obj.store.Lock()`). 
>
> It is exactly as we have it right now, the only difference is that store is moved to its own package.